autogen/agentchat/contrib/vectordb/mongodb.py

```diff
@@ -0,0 +1,287 @@
+            # Create the search index if it doesn't exist
+            search_index_model = SearchIndexModel(
+                definition={
+                    "fields": [
+                        {
+                            "type": "vector",
+                            "numDimensions": self.dimensions,
+                            "path": "embedding",
+                            "similarity": similarity
+                        },
+                    ]
+                },
+                name=index_name,
+                type="vectorSearch"
+            )
+            # Create the search index
+            try:
+                collection.create_search_index(model=search_index_model)
+                return collection
+            except Exception as e:
+                print(f"Error creating search index: {e}")
+                raise e
+        
+    def get_collection(self, collection_name: str = None):
+        """
+        Get the collection from the vector database.
+
+        Args:
+            collection_name: str | The name of the collection. Default is None. If None, return the
+                current active collection.
+        """
+        if collection_name is None:
+            if self.active_collection is None:
+                raise ValueError("No collection is specified.")
+            else:
+                return self.active_collection
+        else:
+            return self.db[collection_name]
+
+    def delete_collection(self, collection_name: str):
+        """
+        Delete the collection from the vector database.
+
+        Args:
+            collection_name: str | The name of the collection.
+        """
+        return self.db.drop_collection(collection_name)
+
+    def insert_docs(self, docs: List[Document], collection_name: str = None, upsert: bool = False):
+        """
+        Insert documents into the collection of the vector database.
+
+        Args:
+            docs: List[Document] | A list of documents. Each document is a TypedDict `Document`.
+            collection_name: str | The name of the collection. Default is None.
+            upsert: bool | Whether to update the document if it exists. Default is False.
+        """
+        if not docs:
+            return
+        if docs[0].get("content") is None:
+            raise ValueError("The document content is required.")
+        if docs[0].get("id") is None:
+            raise ValueError("The document id is required.")
+        collection = self.get_collection(collection_name)
+        for doc in docs:
+            if "embedding" not in doc:
+                doc["embedding"] = np.array(self.embedding_function([
+                    str(doc["content"])
+                ])).tolist()[0]
+                print("embedding")
+                print(doc["embedding"])
+            if upsert:
+                return collection.replace_one({'id': doc['id']}, doc, upsert=True)
+            else:
+                return collection.insert_one(doc)
```

Comment:
I would suggest if upsert=True then doing the one-by-one insertion. 
```suggestion
        for doc in docs:
            if "embedding" not in doc:
                doc["embedding"] = np.array(self.embedding_function([
                    str(doc["content"])
                ])).tolist()[0]
        if upsert:
            for doc in docs:
                return collection.replace_one({'id': doc['id']}, doc, upsert=True)
        else:
            return collection.insert_many(docs)
```

Reply:
Code has been updated. Will push shortly.